### PR TITLE
Hive: Capitalize write format value when creating writer

### PR DESCRIPTION
**Pull request by marton-bod**
Following up on https://github.com/apache/iceberg/issues/2440

### CODE REVIEW

**Comment by marton-bod** [comment]
@pvary @aokolnychyi @RussellSpitzer Can you please take a look?

**Review by pvary — APPROVED** [review]

**Review by RussellSpitzer — APPROVED** [review]
I swear I saw this PR with capitlizing the property in TableProperties, but I like this approach much better

**Comment by marton-bod** [comment]
> I swear I saw this PR with capitlizing the property in TableProperties, but I like this approach much better

You are right, that was my initial commit but then I force pushed this second approach :)

**Comment by pvary** [comment]
Can we add a test case using the default value?

**Comment by marton-bod** [comment]
> Can we add a test case using the default value?

Yes, good idea, added a test case.

File: mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerWithEngine.java [hunk 1]
```diff
@@ -662,6 +663,27 @@ public void testMultiTableInsert() throws IOException {
     HiveIcebergTestUtils.validateData(target2, target2Records, 1);
   }
 
+  @Test
+  public void testWriteWithDefaultWriteFormat() {
+    Assume.assumeTrue("Tez write is not implemented yet", executionEngine.equals("mr"));
```
Comment (pvary) [inline mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerWithEngine.java:668]: Do we need to run this for every catalog / fileFormat?
Maybe we do not want to create a test class just for this, but at least skip the unnecessary tests.

Comment (marton-bod) [reply mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerWithEngine.java:668]: That's right, it doesn't add any value to run it for different configurations. I'm going to expand the Assume.

**Comment by marton-bod** [comment]
I got lots of flaky spark test failures on my latest CI run here:
```
org.apache.iceberg.spark.source.TestIcebergSourceHiveTables3 > classMethod FAILED
    java.net.BindException: Cannot assign requested address: Service 'sparkDriver' failed after 16 retries (on a random free port)! Consider explicitly setting the appropriate binding address for the service 'sparkDriver' (for example spark.driver.bindAddress for SparkDriver) to the correct binding address.
```
Have you seen these before @aokolnychyi @RussellSpitzer ?

**Comment by RussellSpitzer** [comment]
They started cropping up recently ... Not sure what the cause is, perhaps we are failing to turn off a context somewhere? Ie we have 16 Spark Contexts running at this point and then 17 tries to bind and runs out of attempts?

**Review by pvary — APPROVED** [review]

**Comment by pvary** [comment]
Pushed to master. Thanks for the patch @marton-bod and @RussellSpitzer for the review!

**Pull request merged**